NamespaceFixer/NamespaceBuilder/NamespaceBuilderFactory.cs

```diff
@@ -1,32 +1,36 @@
 using NamespaceFixer.Core;
 using System;
+using System.IO;
 
 namespace NamespaceFixer.NamespaceBuilder
 {
     internal class NamespaceBuilderFactory
     {
-        internal static INamespaceBuilder CreateNamespaceBuilderService(string extension, INamespaceAdjusterOptions options)
+        internal static INamespaceBuilder CreateNamespaceBuilderService(string extension, INamespaceAdjusterOptions options, string filePath)
         {
-            INamespaceBuilder rslt = null;
             string projectName = ProjectHelper.GetProjectExtensionName(extension);
+            string fileExtension = Path.GetExtension(filePath);
 
-            switch (projectName)
+            if(projectName == Statics.CsProjectFileExtension)
             {
-                case Statics.CsProjectFileExtension:
-                    rslt = new CsNamespaceBuilderService(options);
-                    break;
-
-                case Statics.VbProjectFileExtension:
-                    rslt = new VbNamespaceBuilderService(options);
-                    break;
+                switch (fileExtension)
+                {
+                    case ".cs":
+                        return new CsNamespaceBuilderService(options);
+                    case ".xaml":
+                        return new XamlNamespaceBuilderService(options);
+                    default:
+                        throw new Exception($"Unsupported file extension '{fileExtension}'.");
+                }
             }
-
-            if (rslt is null)
+            else if(projectName == Statics.VbProjectFileExtension)
+            {
+                return new VbNamespaceBuilderService(options);
+            }
```

Comment:
Should not XAML namespace fixing be supported for VB projects too?

Reply:
After some fiddling in VS. It appears that VB XAML files do not use nested namespaces by default. However if VS Namespace Fixer is used on the linked `.vb` file, this could break the `.xaml` file.

Reply:
I do not know VB language. However it doesn't change behavior for VB projects.

Reply:
![vbcsdiff](https://user-images.githubusercontent.com/24236468/78724628-f9fbeb00-7981-11ea-92a1-2f4c58c7d375.png)

Compare theses two projects with identical structure. Note particularly that in the VB project the files in the nested folder are either still in the root namespace or unlabeled.

My point here is that your code currently only checks for an `.xaml` file extension if the project is a `.csproj`. It needs to check for that in VB projects too. Since despite the different namespace handling, VS Namespace Fixer is able to break those projects by operating on the `.vb` files, it should therefore provide the option to migrate the `.xaml` files too.

Aside from this, the way VB handles namespaces in this instance may be a point to be considered further.

I hope this helps.

Reply:
> I do not know VB language

Neither do I really, these are just my observations from starting a couple of WPF projects in Visual Studio and comparing them. Ideally we need someone familiar with VB to have a look

Reply:
Exactly guys, the implementation that you made is such an improvement to the tool, and you make and amazing work!
If someone needs to work with VB, they can implement it themselves, so they will know exactly what they need